compiler/rustc_trait_selection/src/solve/eval_ctxt.rs

```diff
@@ -285,18 +285,15 @@ impl<'a, 'tcx> EvalCtxt<'a, 'tcx> {
                             // The rhs of this `normalizes-to` must always be an unconstrained infer var as it is
                             // the hack used by `normalizes-to` to ensure that every `normalizes-to` behaves the same
                             // regardless of the rhs.
-                            //
-                            // However it is important not to unconditionally replace the rhs with a new infer var
-                            // as otherwise we may replace the original unconstrained infer var with a new infer var
-                            // and never propagate any constraints on the new var back to the original var.
```

Comment:
I'm not sure if this ever matters in practice?

 Or to put it in another perspective, if this indeed *does* matter, I don't understand why it's then valid to replace a *constrained* RHS with a fresh var here without later equating that constrained RHS against that var.